File: javascript/1260.Shift-2D-Grid.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number[][]}
 */
var shiftGrid = function (grid, k) {
  const rows = grid.length;
  const cols = grid[0].length;
  const total = rows * cols;

  k %= total;

  const result = Array.from({ length: rows }, () => Array(cols));

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      const idx = r * cols + c;
      const newIdx = (idx + k) % total;

      const newR = Math.floor(newIdx / cols);
      const newC = newIdx % cols;

      result[newR][newC] = grid[r][c];
    }
  }

  return result;
};

File: python/1260.Shift-2D-Grid.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        rows = len(grid)
        cols = len(grid[0])
        total = rows * cols
        k %= total

        result = [[None] * cols for _ in range(rows)]

        for r in range(rows):
            for c in range(cols):
                idx = r * cols + c
                new_idx = (idx + k) % total

                new_r = new_idx // cols
                new_c = new_idx % cols

                result[new_r][new_c] = grid[r][c]

        return result